cloudify_cli/commands/sites.py

```diff
@@ -0,0 +1,162 @@
+########
+# Copyright (c) 2013-2019 Cloudify Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+############
+
+
+from .. import env
+from .. import utils
+from ..cli import cfy
+from ..table import print_data, print_single
+from ..utils import handle_client_error, validate_visibility
+
+SITES_COLUMNS = ['name', 'latitude', 'longitude', 'visibility', 'tenant_name',
+                 'created_at', 'created_by']
+
+
+@cfy.group(name='sites')
+@cfy.options.common_options
+def sites():
+    """
+    Handle Cloudify sites
+    """
+    if not env.is_initialized():
+        env.raise_uninitialized()
+
+
+@sites.command(name='create', short_help='Create a new site')
+@cfy.argument('name', callback=cfy.validate_name)
+@cfy.options.latitude
+@cfy.options.longitude
+@cfy.options.visibility(mutually_exclusive_required=False)
+@cfy.options.tenant_name(required=False, resource_name_for_help='site')
+@cfy.options.common_options
+@cfy.assert_manager_active()
+@cfy.pass_client(use_tenant_in_header=True)
+@cfy.pass_logger
+def create(name, latitude, longitude, visibility, tenant_name, client, logger):
+    """Create a new site
+
+    `NAME` is the new site's name
+    """
+    utils.explicit_tenant_name_message(tenant_name, logger)
+    validate_visibility(visibility)
+    client.sites.create(name, latitude, longitude, visibility)
+    logger.info('Site `{0}` created'.format(name))
+
+
+@sites.command(name='get', short_help='Get details for a single site')
+@cfy.argument('name', callback=cfy.validate_name)
+@cfy.options.tenant_name(required=False, resource_name_for_help='site')
+@cfy.options.common_options
+@cfy.assert_manager_active()
+@cfy.pass_client(use_tenant_in_header=True)
+@cfy.pass_logger
+def get(name, tenant_name, client, logger):
+    """Get details for a single site
+
+    `NAME` is the site's name
+    """
+    utils.explicit_tenant_name_message(tenant_name, logger)
+    graceful_msg = 'Requested site with name `{0}` was not found in this ' \
+                   'tenant'.format(name)
+    with handle_client_error(404, graceful_msg, logger):
```

Comment:
I don't think this is needed, without this it will print out the msg from the error from the rest server

Reply:
I think it is nicer : 
```
(cloudify) vagrant@docker-host:~$ cfy sites get site_1
Getting info for site `site_1`...
Requested site with name `site_1` was not found in this tenant
```

Instead of :
```
(cloudify) vagrant@docker-host:~$ cfy sites get site_1
Getting info for site `site_1`...
An error occurred on the server: 404: Requested `Site` with ID `site_1` was not found
```
Also, it may confuse clients to see the `ID` instead of `name`

Reply:
I think it is nicer : 
```
(cloudify) vagrant@docker-host:~$ cfy sites get site_1
Getting info for site `site_1`...
Requested site with name `site_1` was not found in this tenant
```

Instead of :
```
(cloudify) vagrant@docker-host:~$ cfy sites get site_1
Getting info for site `site_1`...
An error occurred on the server: 404: Requested `Site` with ID `site_1` was not found
```
Also, it may confuse clients to see the `ID` instead of `name`

Reply:
So maybe add a new decorator which allow adding a custom error msg, this will remove the duplicate code